okhttp/src/main/java/okhttp3/ConnectionSpec.kt

```diff
@@ -43,51 +43,90 @@ import javax.net.ssl.SSLSocket
  *
  * [tls_history]: https://github.com/square/okhttp/wiki/TLS-Configuration-History
  */
-class ConnectionSpec internal constructor(builder: Builder) {
-  val isTls: Boolean = builder.tls
-  internal val supportsTlsExtensions: Boolean = builder.supportsTlsExtensions
-  internal val cipherSuites: Array<String>? = builder.cipherSuites
-  internal val tlsVersions: Array<String>? = builder.tlsVersions
+class ConnectionSpec internal constructor(
+  @get:JvmName("isTls") val isTls: Boolean,
+  @get:JvmName("supportsTlsExtensions") val supportsTlsExtensions: Boolean,
+  private val cipherSuitesAsString: Array<String>?,
+  private val tlsVersionsAsString: Array<String>?
+) {
+
+  private var lazyCipherSuites: List<CipherSuite>? = null
```

Comment:
This seems problematic, e.g. makes us consider concurrency.  Why bundle this as part of this PR?

Reply:
It’s the same pattern we do in Request.cacheControl, which works. Regardless for consistent behavior with what we did previously this should probably just compute on each call.

Reply:
Removed the cache!